scrapper.py

```diff
@@ -25,39 +36,175 @@ class Crawler:
     """
     Crawler implementation
     """
+
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        """
+        Extract URL
+        """
+        sections_bs = article_bs.find('div', {'class': 'sections'})
+        urls_bs = sections_bs.find_all('a')
+        for url_bs in urls_bs:
+            if len(self.urls) < self.max_articles:
+                if url_bs['href'] not in self.urls:
+                    self.urls.append(url_bs['href'])
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+
+        for seed_url in self.seed_urls:
+            response = requests.get(seed_url, headers=HEADERS)
+            if not response.ok:
+                print("Failed")
+
+            article_bs = BeautifulSoup(response.text, 'lxml')
+            self._extract_url(article_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class HTMLParser:
+    """
+    HTMLWithPDFParser implementation
+    """
+
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(self.article_url, self.article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        """
+        Extract text
+        """
+        body_bs = article_bs.find('body')
+        links_bs = body_bs.find_all('a')
+        print(links_bs)
+        # print(article_bs)
+        # galley = article_bs.find('div', {'class': 'item galleys'})
+        # print(galley)
+        pdf_link = ''
+        try:
```

Comment:
using try with such a huge body is not recommended. remember to surround only vulnerable piece of code with try/except. I can hardly imagine how each line produces SchemaError. however, this is how your code now looks like.

narrow down the vulnerable code